RobotControl/Robot.cpp

```diff
@@ -50,41 +61,115 @@ void Robot::update() {
 
   int timeLeftMillis = TICK_DURATION_MILLIS - (millis() - tickStartMillis);
   if (timeLeftMillis > 0) {
-    delay(timeLeftMillis);
+    //delay(timeLeftMillis);
   }
 }
 
 void Robot::updateSerial() {
-  if (Serial.available() > 0) {
-    String line = Serial.readString();
-    updateFromJson(line.c_str());
-    m_state.printJson(Serial);
-    Serial.println();
+  memset(m_payloadBytes, 0, PAYLOAD_LEN);
+  memset(m_serialBuffer, 0, SERIAL_BUFFER_LEN);
+
+  if (m_commsI2C.getBytes(m_serialBuffer, SERIAL_BUFFER_LEN, m_payloadBytes, PAYLOAD_LEN)) {
+    updatePayload(m_payloadBytes, PAYLOAD_LEN);
+    //Serial.println("Bytes parsed");
   }
+
+  setStatus();
+  setRobot();
+  
+  //m_payload.print();
+  //Serial.println();
 }
 
-void Robot::updateFromJson(const char *json) {
-  m_state.updateFromJson(json);
+void Robot::updatePayload(const uint8_t *data, const uint8_t len) {
+  bool success = m_payload.readMessage(data, len);
 
-  const char *status = m_state.root()["status"].asString();
-  const char *err = m_state.root()["err"].asString();
+  const uint8_t status = m_payload.getStatus();
+  const uint8_t err = m_payload.getError();
 
-  if (strlen(err) > 0) {
+  if (err > 0 || !success) {
     m_statusLEDs.setBlinkPattern(StatusLEDs::ERROR);
-    m_state.root()["status"] = STATUS_DISABLED;
+    m_payload.setStatus(STATUS_DISABLED);
   } else {
-    if (strcmp(status, STATUS_DISABLED) == 0) {
+    if (status == STATUS_DISABLED) {
       m_statusLEDs.setBlinkPattern(StatusLEDs::DISABLED);
-    } else if (strcmp(status, STATUS_ENABLED) == 0) {
+    } else if (status == STATUS_ENABLED) {
       m_statusLEDs.setBlinkPattern(StatusLEDs::ENABLED);
-    } else if (strcmp(status, STATUS_PRIMED) == 0) {
+    } else if (status == STATUS_PRIMED) {
       m_statusLEDs.setBlinkPattern(StatusLEDs::PRIMED);
     } else {
       m_statusLEDs.setBlinkPattern(StatusLEDs::OFF);
     }
   }
 }
 
+void Robot::setRobot() {
```

Comment:
Ideally, this would take the payload as an argument instead of using a local member. Otherwise I'd recommend naming this `updateRobot` instead of `setRobot`